api/document/serializers/doc_cursor.py

```diff
@@ -25,22 +65,17 @@ class Meta:
             'title',
             'type_emblem',
         )
+        read_only_fields = tuple(
+            field for field in fields
+            if field not in PRIMITIVE_DOC_NODE_FIELDS + [
+                'children',
+                'content'
+            ]
+        )
 
-    def get_children(self, instance):
-        return type(self)(
-            instance.children(), many=True,
-            context={**self.context, 'is_root': False},
-        ).data
-
-    def get_content(self, instance):
-        """Include all annotations of the text."""
-        annotations = nest_annotations(
-            instance.annotations(), len(instance.text))
-        return NestedAnnotationSerializer(
-            annotations,
-            context={'cursor': instance, 'parent_serializer': self},
-            many=True,
-        ).data
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.context['parent_serializer'] = self
```

Comment:
This feels weird, but I couldn't find any easier/cleaner way to pass the parent serializer to the new `ContentField` class.

Reply:
I bet there's a static field we can override (DRF seems to love this pattern) for "child context" or something similar. No need to find it now, but worth investigating.